cli/src/main/java/software/amazon/smithy/java/cli/SmithyCall.java

```diff
@@ -0,0 +1,288 @@
+        String defaultArnNamespace = serviceInput.getNamespace().toLowerCase();
+        if (auth != null) {
+            switch (auth.authType.toLowerCase()) {
+                case "sigv4":
+                case "aws":
+                    if (auth.awsRegion == null) {
+                        throw new IllegalArgumentException("SigV4 auth requires --aws-region to be set. Please provide the --aws-region option.");
+                    }
+                    builder.putConfig(RegionSetting.REGION, auth.awsRegion)
+                            .putSupportedAuthSchemes(new SigV4AuthScheme(defaultArnNamespace))
+                            .authSchemeResolver(AuthSchemeResolver.DEFAULT)
+                            .addIdentityResolver(new EnvironmentVariableIdentityResolver());
+                    break;
+                case "none":
+                    builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported auth type: " + auth.authType);
+            }
+        } else {
+            builder.authSchemeResolver(AuthSchemeResolver.NO_AUTH);
+        }
+    }
+
+    private void configureProtocol(DynamicClient.Builder builder, ShapeId serviceInput) {
+        if (protocolType != null) {
+            builder.protocol(switch (protocolType) {
+                case AWS_JSON -> new AwsJson1Protocol(serviceInput);
+                case RPC_V2_CBOR -> new RpcV2CborProtocol(serviceInput);
+                case REST_JSON -> new RestJsonClientProtocol(serviceInput);
+                case REST_XML -> new RestXmlClientProtocol(serviceInput);
+            });
+        }
+    }
+
+    private void configureInputInterceptor(DynamicClient.Builder builder) {
+        if (inputJson != null || inputPath != null) {
+            builder.addInterceptor(new ClientInterceptor() {
+                @Override
+                public void readBeforeTransmit(RequestHook<?, ?, ?> hook) {}
+            });
+        }
+    }
+
+    private Document executeClientCall(DynamicClient client) throws Exception {
+        if (inputJson != null && inputPath != null) {
+            throw new IllegalArgumentException("Cannot specify both '--input-json' and '--input-path'. Please provide only one.");
+        }
+
+        LOGGER.fine("Executing client call");
+
+        if (inputJson == null && inputPath == null) {
+            return client.call(operation);
+        }
+
+        Document inputDocument;
+        if (inputJson != null) {
+            inputDocument = CODEC.createDeserializer(inputJson.getBytes(StandardCharsets.UTF_8)).readDocument();
+        } else {
+            String content = Files.readString(Path.of(inputPath), StandardCharsets.UTF_8);
+            inputDocument = CODEC.createDeserializer(content.getBytes(StandardCharsets.UTF_8)).readDocument();
+        }
+
+        return client.call(operation, inputDocument);
```

Comment:
1. You can avoid first converting to a string and then back to bytes when reading from a file.
2. You should close the deserializer.

```suggestion
        byte[] inputBytes;
        if (inputJson != null) {
            inputBytes = inputJson.getBytes(StandardCharsets.UTF_8);
        } else {
            inputBytes = Files.readAllBytes(Path.of(inputPath));
        }

        Document input;
        try (var d = CODEC.createDeserializer(inputBytes)) {
            input = d.readDocument();
        }

        return client.call(operation, input);
```

Reply:
I decided to keep `if (inputJson == null && inputPath == null) {
            return client.call(operation);
        }` to differentiate calls with input vs calls without input. Implemented rest of the changes.